extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByQueryKit.java

```diff
@@ -185,13 +185,14 @@ public QueryDefinition makeQueryDefinition(
       );
 
       if (doLimitOrOffset) {
+        final ShuffleSpec finalShuffleSpec = resultShuffleSpecFactory.build(resultClusterBy, false);
```

Comment:
Assume select q's are group by q's for this comment : 
https://github.com/apache/druid/pull/16643#discussion_r1665416378